contracts/ZkDai.sol

```diff
@@ -0,0 +1,153 @@
+pragma solidity ^0.4.25;
+
+import 'openzeppelin-solidity/contracts/token/ERC20/ERC20.sol';
+import "./MintNotes.sol";
+import "./SpendNotes.sol";
+
+contract ZkDai is MintNotes, SpendNotes {
+  ERC20 internal DAI_TOKEN_ADDRESS;
```

Comment:
Did you mean why _not_ the constant typing? Yes, the variable should be a constant with address set to the current dai contract. I didn't find a network agnostic way of setting the variable - it needs to be hardcoded according to the network being deployed to. I am passing it in the constructor because the tests are deploying a mock dai contract on the go.

But yes ideally, before deploying to mainnet this variable should be a `constant`.

Reply:
I think it is fine for it to be a state variable that can only be set in the constructor, but i would make it `ERC20 public dai`

Reply:
Got it. I misunderstood the comment.